pkg/nodeidentity/interfaces.go

```diff
@@ -27,5 +27,6 @@ type Identifier interface {
 }
 
 type Info struct {
-	InstanceGroup string
+	InstanceGroup     string
+	InstanceLifecycle string
```

Comment:
In this case, a string it's not really needed to check the value. I think it is important only if it is a spot instance or not? 
```suggestion
	InstanceSpot bool
```

Reply:
> In this case, a string it's not really needed to check the value. I think it is important only if it is a spot instance or not?

True! The string type made sense in the first approach. I'll update the PR when I find time.